File: backend/src/main/java/com/back/ourlog/external/spotify/dto/TrackItem.java
package com.back.ourlog.external.spotify.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.List;

@Getter
@NoArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true)
public class TrackItem {
    private String id;
    private String name;
    private List<SpotifyArtist> artists;
    private SpotifyAlbum album;
    private int popularity;

    @JsonProperty("external_urls")
    private ExternalUrls externalUrls;
}

File: backend/src/main/java/com/back/ourlog/external/spotify/dto/TrackItem.kt
package com.back.ourlog.external.spotify.dto

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

@JsonIgnoreProperties(ignoreUnknown = true)
data class TrackItem(
    val id: String? = null,
    val name: String? = null,
    val artists: List<SpotifyArtist?>? = null,
    val album: SpotifyAlbum? = null,
    val popularity: Int = 0,

    @JsonProperty("external_urls")
    val externalUrls: ExternalUrls? = null
)
